### PR TITLE
[NP-4224] Duplicate email some with missing subject

### PR DESCRIPTION
**Pull request by Mykola97**
https://nanopay.atlassian.net/browse/NP-4224

### CODE REVIEW

**Review by jlhughes — COMMENTED** [review]
Looks good. 
Where are the exceptions caught and reported? And when reported is there enough info to track them down?

**Comment by Mykola97** [comment]
> Looks good.
> Where are the exceptions caught and reported? And when reported is there enough info to track them down?
The exceptions got caught inside EmailsUtility and then logged.
Should be enough to track the error down.

Screen Shot 2021-04-27 at 2 49 18 PM

**Comment by jlhughes** [comment]
Can you see if it's possible to report the name of the email template, if one is used.

**Comment by Mykola97** [comment]
> Can you see if it's possible to report the name of the email template, if one is used.

Added

**Review by Nauna — APPROVED** [review]
This looks really nice.

File: src/foam/util/Emails/EmailsUtility.java [hunk 1]
```diff
@@ -137,7 +137,8 @@ public static void sendEmailFromTemplate(X x, User user, EmailMessage emailMessa
     try {
       cts.apply(userX, group, emailMessage, templateArgs);
     } catch (Exception e) {
-      logger.error(e);
+
+      logger.error("Problem with template: " + templateName, e);
```
Comment (jlhughes) [inline src/foam/util/Emails/EmailsUtility.java:141]: Please also add an alarm here - warning is fine. 
name: EmailTemplate
note: add the templateName

**Pull request merged**